### PR TITLE
[SPARK-4127] [MLlib] [PySpark] Python bindings for StreamingLinearRegressionWithSGD

### PR DESCRIPTION
**Pull request by MechCoder**
Python bindings for StreamingLinearRegressionWithSGD

### CODE REVIEW

**Comment by MechCoder** [comment]
cc: @freeman-lab Will remove the LinearDataGenerator stuff (or rebase over master) as soon as https://github.com/apache/spark/pull/6715 is merged.

File: python/pyspark/mllib/regression.py [hunk 1]
```diff
@@ -570,6 +571,92 @@ def train(cls, data, isotonic=True):
         return IsotonicRegressionModel(boundaries.toArray(), predictions.toArray(), isotonic)
 
 
+@inherit_doc
+class StreamingLinearRegressionWithSGD(LinearRegressionModel):
+    """
+    Run LinearRegression with SGD on a stream of data.
+
+    The problem minimized is (1 / n_samples) * (y - weights'X)**2.
+    After training on a stream of data, the weights obtained at the end of
```
Comment (freeman-lab) [inline python/pyspark/mllib/regression.py]: `stream` -> `batch`

File: python/pyspark/mllib/regression.py [hunk 2]
```diff
@@ -570,6 +571,92 @@ def train(cls, data, isotonic=True):
...
+    training are used as initial weights for the next stream of data.
```
Comment (freeman-lab) [inline python/pyspark/mllib/regression.py]: `stream` -> `batch`

File: python/pyspark/mllib/regression.py [hunk 3]
```diff
@@ -570,6 +571,92 @@ def train(cls, data, isotonic=True):
...
+                "dstream should be a DStream object, got %s" % type(dstream))
+        if not self._model:
+            raise ValueError(
+                "Model must be intialized using setInitialWeights")
+
+    @property
+    def latestModel(self):
+        """Returns a LinearRegressionModel fit on the latest stream of data.
+
+        The weights and intercepts can be got from the `weights` and
```
Comment (freeman-lab) [inline python/pyspark/mllib/regression.py]: `got` -> `obtained`

File: docs/mllib-linear-methods.md [hunk 4]
```diff
@@ -768,6 +768,58 @@ will get better!
...
+
+{% highlight python %}
+numFeatures = 3
+model = StreamingLinearRegressionWithSGD()
+model.setInitialWeights([0.0, 0.0, 0.0])
+{% endhighlight %}
+
+Now we register the streams for training and testing and start the job.
+
+{% highlight scala %}
```
Comment (freeman-lab) [inline docs/mllib-linear-methods.md]: language should be python right?

**Comment by sujkh85** [comment]
NAVER - http://www.naver.com/
--------------------------------------------

sujkh@naver.com 님께 보내신 메일 <Re: [spark] [SPARK-4127] [MLlib] [PySpark] Python bindings for StreamingLinearRegressionWithSGD (#6744)> 이 다음과 같은 이유로 전송 실패했습니다.

--------------------------------------------

받는 사람이 회원님의 메일을 수신차단 하였습니다. 


--------------------------------------------

File: docs/mllib-linear-methods.md (hunk 4 above)
Comment (MechCoder) [inline docs/mllib-linear-methods.md]: Forgot to change while copy pasting :P

**Comment by MechCoder** [comment]
@freeman-lab @mengxr I rebased over master and inherited from `StreamingLinearAlgorithm` to avoid code duplication. Please give another pass.

**Comment by MechCoder** [comment]
I moved `StreamingLinearAlgorithm` to `regression.py` so that it complies with `LabeledPoint` and (there were circular imports)

**Comment by MechCoder** [comment]
jenkins retest this please

**Comment by mengxr** [comment]
LGTM. Merged into master. Thanks! (The last Jenkins build was successful.)

**Pull request closed without merging**